CommonTools/ConditionDBWriter/interface/ConditionDBWriter.h

```diff
@@ -358,10 +358,13 @@ class ConditionDBWriter : public edm::EDAnalyzer {
       objPointer = getNewObject();
       
       if (!objPointer ) {
-	edm::LogError("ConditionDBWriter::storeOnDbNow: ERROR - requested to store on DB a new object (module configuration is algo driven based IOV), but received NULL pointer...will not store anything on the DB") << std::endl;
-	return;
+        edm::LogError("ConditionDBWriter::storeOnDbNow: ERROR - requested to store on DB a new object (module configuration is algo driven based IOV), but received NULL pointer...will not store anything on the DB") << std::endl;
+        return;
+      }
+      else {
+        storeOnDb(objPointer);
+        delete objPointer;
```

Comment:
It looks that this deletion policy is not applied everywhere in ConditionDBWriter. Please make it consistent in all the concerned methods ( endRun and endLuminosityBlock ). Also, consider the use of std::unique_ptr